.design/requirements-575.md

```diff
@@ -0,0 +1,269 @@
+# Jakarta WebSocket Whiteboard
+
+This documents looks into the requirements for a Jakarta WebSocket whiteboard specification.
+
+## Introduction
+
+The OSGi Compendium R6 introduced the HTTP Whiteboard specification, upgraded in R8.1 to the Whiteboard specification for Jakarta Servlet.
+It provides a light and convenient way of using servlets in a an OSGi environment through the use of the Whiteboard Pattern.
+
+WebSocket is communication protocol designed to work over the HTTP protocol.
+It enables bidirectional exchange of messages between the web client and the server.
+Like an HTTP servlet, a WebSocket handler is bound to a URL.
+
+The conversion from an HTTP connection to a WebSocket stream is done by sending a `GET` HTTP request with a set of headers requesting a connection upgrade.
```

Comment:
https://www.rfc-editor.org/rfc/rfc6455


1.1
```
The WebSocket Protocol is designed to supersede existing
   bidirectional communication technologies that use HTTP as a transport
   layer to benefit from existing infrastructure (proxies, filtering,
   authentication).  Such technologies were implemented as trade-offs
   between efficiency and reliability because HTTP was not initially
   meant to be used for bidirectional communication (see [[RFC6202](https://www.rfc-editor.org/rfc/rfc6202)] for
   further discussion).  The WebSocket Protocol attempts to address the
   goals of existing bidirectional HTTP technologies in the context of
   the existing HTTP infrastructure; as such, it is designed to work
   over HTTP ports 80 and 443 as well as to support HTTP proxies and
   intermediaries, even if this implies some complexity specific to the
   current environment.  However, the design does not limit WebSocket to
   HTTP, and future implementations could use a simpler handshake over a
   dedicated port without reinventing the entire protocol.  This last
   point is important because the traffic patterns of interactive
   messaging do not closely match standard HTTP traffic and can induce
   unusual loads on some components.
   ```

Reply:
just wanted to point to the position in original rfc